src/main/java/lotto/domain/LottoTicket.java

```diff
@@ -2,29 +2,59 @@
 
 import java.util.ArrayList;
 import java.util.Collections;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
+import java.util.function.Predicate;
 
 import lotto.domain.strategy.GenerateStrategy;
 
 public class LottoTicket {
-	private static final Integer NUMBERS_COUNT = 6;
-	private final Set<Integer> numbers;
-
-	public LottoTicket(GenerateStrategy strategy) {
-		this.numbers = strategy.generate();
-		isNumbersSizeEqCount();
-	}
-
-	public List<Integer> getNumbers() {
-		List<Integer> numberList = new ArrayList<>(numbers);
-		Collections.sort(numberList);
-		return Collections.unmodifiableList(numberList);
-	}
-
-	private void isNumbersSizeEqCount() {
-		if (numbers.size()!= NUMBERS_COUNT) {
-			throw new IllegalArgumentException("올바르지 않은 추첨 결과입니다.");
-		}
-	}
+    private Set<LottoNumber> numbers;
+
+    public LottoTicket(GenerateStrategy strategy) {
+        numbers = new LinkedHashSet<>();
+        this.numbers = strategy.generate();
+        isNumbersSizeEqCount();
+    }
+
+    public LottoTicket(Set<Integer> nums) {
+        numbers = new LinkedHashSet<>();
+        for (Integer num : nums) {
+            numbers.add(new LottoNumber(num));
+        }
+        isNumbersSizeEqCount();
+    }
+
+    public LottoTicket(int... nums) {
+        numbers = new LinkedHashSet<>();
+        for (int num : nums) {
+            this.numbers.add(new LottoNumber(num));
+        }
+        isNumbersSizeEqCount();
```

Comment:
```suggestion
        this(Set.of(nums));
```

이걸로 대체할 수 있지 않을까요?
다른 생성자도 코드 중복을 제거할 수 있을 것 같아요.

Reply:
해당 부분은 Set<Integer>를 Set<LottoNumber>로 변환해주는 부분인데, 어떤 방법으로 Set.of를 활용하여 리팩토링해야할지 잘 모르겠습니다. Set.of(new LottoNumber(nums))와 같이 LottoNumber내 생성자를 만들어 변환해주는 방법이 있긴하지만 그렇게 된다면 책임 분리 측면에서 좋지 못하다고 생각했습니다.

Reply:
생성자 `LottoTicket(Set<Integer> nums)` , `LottoTicket(int... nums)` 의 내용은 완전히 같은 것으로 보입니다.
또한 지금의 도메인에서는 생성자의 내용이 별도로 바뀔 것 같지도 않구요.

```java
    public LottoTicket(int... nums) {
        this(Set.of(nums));
    }
```

이 코드는 역할이 비슷한 다른 생성자로 옮겨 수행 역할을 할 뿐, 책임 분리와는 상관이 없지 않을까요?
생성자의 책임인 생성에 충실하며 중복 코드를 제거했을 뿐입니다.

코드의 내용이 완전 비슷하나 2벌, 3벌이 중복으로 존재한다면 하나로 통일하여 관리할 때 인적 오류를 줄일 수 있어요.
만약 요구사항이 바뀌어서 생성자 2개 중 1개를 수정했는데 실수로 남은 1개를 수정하지 않는다면 버그의 여지를 남겨두게 되거든요.